### PR TITLE
Enforce circuit/service id formats

### PR DESCRIPTION
**Pull request by ltseeley**
Updates the admin service's circuit validation to check that the circuit
and service IDs of proposed circuits are valid.

Also updates gameroom to use valid circuit and service IDs.

To test, run gameroom with `docker-compose -f examples/gameroom/docker-compose.yaml up --build` and verify that you can create a gameroom successfully.

### CODE REVIEW

File: examples/gameroom/daemon/src/rest_api/routes/gameroom.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ use gameroom_database::{
 use openssl::hash::{hash, MessageDigest};
 use protobuf::Message;
 use splinter::admin::messages::{
-    CreateCircuit, CreateCircuitBuilder, SplinterNode, SplinterService,
+    CreateCircuit, CreateCircuitBuilder, SplinterNode, SplinterServiceBuilder,
```
Comment (agunde406) [inline examples/gameroom/daemon/src/rest_api/routes/gameroom.rs]: Typo in commit messages grXX -> gmXX

File: examples/gameroom/daemon/src/authorization_handler/mod.rs [hunk 2]
```diff
@@ -1323,7 +1326,7 @@ mod test {
     fn get_new_gameroom_service(circuit_id: &str, timestamp: SystemTime) -> NewGameroomService {
         NewGameroomService {
             circuit_id: circuit_id.to_string(),
-            service_id: "scabbard_123".to_string(),
+            service_id: "gr00".to_string(),
```
Comment (agunde406) [inline examples/gameroom/daemon/src/authorization_handler/mod.rs]: should these be gm based on the previous commit?

File: examples/gameroom/daemon/src/rest_api/routes/gameroom.rs (hunk 1 above)
Comment (ltseeley) [reply examples/gameroom/daemon/src/rest_api/routes/gameroom.rs]: Should be `gr` (for gameroom). Fixed the code.

File: examples/gameroom/daemon/src/authorization_handler/mod.rs (hunk 2 above)
Comment (ltseeley) [reply examples/gameroom/daemon/src/authorization_handler/mod.rs]: Fixed previous commit (should be `gr`)

**Pull request merged**